konduit-serving-vertx/src/main/java/ai/konduit/serving/vertx/api/DeployKonduitServing.java

```diff
@@ -0,0 +1,121 @@
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class DeployKonduitServing {
+    public static final String SERVICE_PREFIX = "konduit";
+    public static final String INFERENCE_SERVICE_IDENTIFIER = SERVICE_PREFIX + ":ai.konduit.serving:inference";
+
+    public static Vertx deploy(VertxOptions vertxOptions,
+                              DeploymentOptions deploymentOptions,
+                              InferenceConfiguration inferenceConfiguration,
+                              Handler<AsyncResult<InferenceDeploymentResult>> eventHandler) {
+        Vertx vertx = Vertx.vertx(vertxOptions
+                .setMaxEventLoopExecuteTime(10)
+                .setMaxEventLoopExecuteTimeUnit(TimeUnit.SECONDS));
+        vertx.registerVerticleFactory(new VerticleFactory() {
+            @Override
+            public String prefix() {
+                return SERVICE_PREFIX;
+            }
+
+            @Override
+            public Verticle createVerticle(String verticleName, ClassLoader classLoader) throws Exception {
+                    String protocolName = verticleName.substring(verticleName.lastIndexOf(':') + 1);
+
+                    try {
+                        switch (ServerProtocol.valueOf(protocolName.toUpperCase())) {
+                            case HTTP:
+                                return createVerticleFromClassName("ai.konduit.serving.vertx.protocols.http.verticle.InferenceVerticleHttp");
+                            case GRPC:
+                                return createVerticleFromClassName("ai.konduit.serving.vertx.protocols.grpc.verticle.InferenceVerticleGrpc");
+                            case MQTT:
+                                return createVerticleFromClassName("ai.konduit.serving.vertx.protocols.mqtt.verticle.InferenceVerticleMqtt");
+                            default:
+                                throw new IllegalStateException(
+                                        String.format("Invalid service type %s. Possible values are: %s",
+                                                protocolName,
+                                                Arrays.toString(ServerProtocol.values())
+                                        )
+                                );
+                        }
+                    } catch (ClassNotFoundException classNotFoundException) {
+                        // todo: write plus log a meaningful message and throw exception
+                        throw new IllegalStateException();
+                    }
+            }
+
+            private Verticle createVerticleFromClassName(String className) throws Exception {
+                return (Verticle) ClassLoader.getSystemClassLoader()
+                        .loadClass(className)
+                        .getConstructor().newInstance();
+            }
+        });
+
+        JsonObject jsonConfiguration = new JsonObject(inferenceConfiguration.toJson());
```

Comment:
Let's add a try/catch around inferenceConfiguration.toJson() - normally this shouldn't fail, but it can for pipelines with custom steps - or if the current classpath doesn't contain the module necessary for running this configuration.
(for example, TF step in JSON without the konduit-serving-tensorflow module)